policy/policy_test.go

```diff
@@ -131,3 +135,231 @@ func TestPolicyRead(t *testing.T) {
+	}
+	if _, ok := enabledChecks[checkName]; !ok {
+		t.Error("Expected the check to be added to enabled checks")
+	}
+
+	// Try enabling a check that does not exist
+	nonExistentCheck := "Non-Existent-Check"
+	result = enableCheck(nonExistentCheck, &enabledChecks)
+
+	// Assert the result
+	if result {
+		t.Error("Expected the check to not be enabled")
+	}
+	if _, ok := enabledChecks[nonExistentCheck]; ok {
+		t.Error("Expected the check to not be added to enabled checks")
+	}
+}
+
+func TestIsSupportedCheck(t *testing.T) {
+	t.Parallel()
+
+	// Create a sample check name
+	checkName := "Binary-Artifacts"
+
+	// Create a sample list of required request types
+	requiredRequestTypes := []checker.RequestType{
+		checker.FileBased,
+	}
+
+	// Call the function being tested
+	result := isSupportedCheck(checkName, requiredRequestTypes)
+
+	// Assert the result
+	expectedResult := false
+	if result != expectedResult {
+		t.Errorf("Unexpected result: got %v, want %v", result, expectedResult)
+	}
+
+	// Try with an unsupported check
+	unsupportedCheckName := "Unsupported-Check"
+	result = isSupportedCheck(unsupportedCheckName, requiredRequestTypes)
+
+	// Assert the result
+	expectedResult = false
+	if diff := cmp.Diff(result, expectedResult); diff != "" {
+		t.Errorf("Unexpected result (-got +want):\n%s", diff)
+	}
+
+	// Additional test cases can be added to cover more scenarios
+}
+
+func TestParseFromFile(t *testing.T) {
+	t.Parallel()
+
+	// Provide the path to the policy file
+	policyFile := "testdata/policy-ok.yaml"
+
+	// Call the function being tested
+	sp, err := ParseFromFile(policyFile)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	if len(sp.Policies) != 3 {
+		t.Errorf("Unexpected number of policies: got %v, want %v", len(sp.Policies), 3)
+	}
+	invalidPolicy := "testdata/policy-invalid-score-0.yaml"
+	_, err = ParseFromFile(invalidPolicy)
+	if err == nil {
+		t.Error("Expected an error")
+	}
+	invalidMode := "testdata/policy-invalid-mode.yaml"
+	_, err = ParseFromFile(invalidMode)
+	if err == nil {
+		t.Error("Expected an error")
+	}
+	invalidFile := "testdata/invalid-file.yaml"
+	_, err = ParseFromFile(invalidFile)
+	if err == nil {
+		t.Error("Expected an error")
+	}
+}
+
+func TestModeToProto(t *testing.T) {
+	t.Parallel()
+
+	// Call the function being tested
+	mode := modeToProto("enforced")
+
+	// Check the result
+	expectedMode := CheckPolicy_ENFORCED
+	if mode != expectedMode {
+		t.Errorf("Unexpected mode. Got: %v, Want: %v", mode, expectedMode)
+	}
+
+	// Call the function again with a different mode
+	mode = modeToProto("disabled")
+
+	// Check the result
+	expectedMode = CheckPolicy_DISABLED
+	if mode != expectedMode {
+		t.Errorf("Unexpected mode. Got: %v, Want: %v", mode, expectedMode)
+	}
+
+	// Test panic with an unknown mode
+	testPanic := func() {
+		defer func() {
+			if r := recover(); r == nil {
+				t.Errorf("Expected panic, but no panic occurred")
+			}
+		}()
+		modeToProto("unknown")
```

Comment:
Unrelated to this change (feel free to resolve) but having modeToProto panic seems wrong.

Reply:
I agree. Another PR.